src/btree/bt_discard.c

```diff
@@ -73,9 +73,9 @@ __wt_page_out(WT_SESSION_IMPL *session, WT_PAGE **pagep)
     if (F_ISSET(session->dhandle, WT_DHANDLE_DEAD) || F_ISSET(S2C(session), WT_CONN_CLOSING))
         __wt_page_modify_clear(session, page);
 
-    /* Assert we never discard a dirty page or a page queue for eviction. */
-    WT_ASSERT(session, !__wt_page_is_modified(page));
-    WT_ASSERT(session, !F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU));
+    /* Assert we never discard a dirty page or a page queued for eviction. */
+    WT_ASSERT_STRONG(session, !__wt_page_is_modified(page));
```

Comment:
These asserts have also been updated as they were [triggering in our investigation of WT_9323](https://jira.mongodb.org/browse/WT-9417?focusedCommentId=4595986&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4595986)